xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp

```diff
@@ -263,16 +208,24 @@ CDVDVideoCodec::VCReturn CAddonVideoCodec::GetPicture(VideoPicture* pVideoPictur
 //    pVideoPicture->iLineSize[0] = picture.stride[0];
 //    pVideoPicture->iLineSize[1] = picture.stride[1];
 //    pVideoPicture->iLineSize[2] = picture.stride[2];
-//    pVideoPicture->iWidth = picture.width;
-//    pVideoPicture->iHeight = picture.height;
-//    pVideoPicture->pts = DVD_NOPTS_VALUE;
-//    pVideoPicture->dts = DVD_NOPTS_VALUE;
-//    pVideoPicture->color_range = 0;
-//    pVideoPicture->color_matrix = 4;
-//    pVideoPicture->iFlags = DVP_FLAG_ALLOCATED;
-//    if (m_codecFlags & DVD_CODEC_CTRL_DROP)
-//      pVideoPicture->iFlags |= DVP_FLAG_DROPPED;
-//    pVideoPicture->format = RENDER_FMT_YUV420P;
+    pVideoPicture->iWidth = picture.width;
+    pVideoPicture->iHeight = picture.height;
+    pVideoPicture->pts = picture.pts;
+    pVideoPicture->dts = DVD_NOPTS_VALUE;
+    pVideoPicture->color_range = 0;
+    pVideoPicture->color_matrix = 4;
+    pVideoPicture->iFlags = 0;
+    if (m_codecFlags & DVD_CODEC_CTRL_DROP)
+      pVideoPicture->iFlags |= DVP_FLAG_DROPPED;
+
+    m_videoBufferManager.Configure(AV_PIX_FMT_YUV420P, picture.width, picture.height, picture.stride[0], picture.height, picture.decodedDataSize);
```

Comment:
This is the first point in time that picture.stride[0] is filled in.
It is undefined when GetBuffer is called (https://github.com/xbmc/xbmc/pull/12417/commits/ceced4c3f34a4ac04f8392978f0d63e967853c47#diff-7c4d93a31a66a47a05f4bda3208d9edbR310)
@peak3d will have to explain if there is any way round this.
I did initially request getting the width/height/stride when GetBuffer was called but that was not possible with the API widevine provides.